Xplat/src/main/java/vazkii/botania/common/block/dispenser/GrassSeedsBehavior.java

```diff
@@ -0,0 +1,36 @@
+package vazkii.botania.common.block.dispenser;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.BlockSource;
+import net.minecraft.core.dispenser.OptionalDispenseItemBehavior;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.block.DispenserBlock;
+
+import org.jetbrains.annotations.NotNull;
+
+import vazkii.botania.common.item.GrassSeedsItem;
+import vazkii.botania.network.EffectType;
+import vazkii.botania.network.clientbound.BotaniaEffectPacket;
+import vazkii.botania.xplat.XplatAbstractions;
+
+public class GrassSeedsBehavior extends OptionalDispenseItemBehavior {
+	@NotNull
+	@Override
+	public ItemStack execute(BlockSource source, ItemStack stack) {
+		ServerLevel world = source.getLevel();
+		BlockPos pos = source.getPos().relative(source.getBlockState().getValue(DispenserBlock.FACING));
+
+		setSuccess(((GrassSeedsItem) stack.getItem()).applySeeds(world, pos, stack));
+
+		if (isSuccess()) {
+			XplatAbstractions.INSTANCE.sendToNear(world, pos,
+					new BotaniaEffectPacket(EffectType.GRASS_SEED_PARTICLES,
+							(double) pos.getX(), (double) pos.getY(), (double) pos.getZ(),
+							GrassSeedsItem.getID(((GrassSeedsItem) stack.getItem()).getIslandType(null))));
```

Comment:
why null and not `stack`?

Reply:
Because the argument is unused, and I don't know why, semantically, it ought to be `stack` - I'm not honestly sure what `getIslandType` is supposed to do with its argument

Reply:
Please pass stack then, just because it's unused now doesn't mean you should pass something meaningless like null, because it can start to be used in the future.